commit cac1002f7818d73e2b948fffff087c3d4278b920 (refs/changes/31/1731/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-31T22:47:33+00:00 (2 years, 6 months ago)
    
    CNS-199 cnapi can return non-null, non-string "last_heartbeat" values
    Reviewed by: Richard Kiene <richard.kiene@joyent.com>

diff --git a/lib/cn-filter.js b/lib/cn-filter.js
index 3791d14..88f4258 100644
--- a/lib/cn-filter.js
+++ b/lib/cn-filter.js
@@ -142,7 +142,7 @@ function cutServerObj(obj) {
 	 * Don't bother looking for a heartbeat from a server that isn't
 	 * set up yet. These are always "down".
 	 */
-	if (obj.setup === false || obj.last_heartbeat === null) {
+	if (obj.setup === false || typeof (obj.last_heartbeat) !== 'string') {
 		cutObj.down = true;
 		return (cutObj);
 	}
